src/overlays/actors/ovl_Boss_Dodongo/z_boss_dodongo.c

```diff
@@ -51,311 +44,1717 @@ const ActorInit Boss_Dodongo_InitVars = {
+    ICHAIN_F32_DIV1000(gravity, -3000.0f, ICHAIN_CONTINUE),
+    ICHAIN_F32(targetArrowOffset, 8200.0f, ICHAIN_STOP),
 };
 
-static ColliderJntSphInit D_808C73AC = {
-    {
-        COLTYPE_HIT3,
-        AT_ON | AT_TYPE_ENEMY,
-        AC_ON | AC_TYPE_PLAYER,
-        OC1_ON | OC1_TYPE_PLAYER,
-        OC2_TYPE_1,
-        COLSHAPE_JNTSPH,
-    },
-    19,
-    D_808C7100,
+extern SkeletonHeader D_0601B310;
+extern AnimationHeader D_0600F0D8;
+extern u64 D_030021D8[];
+extern AnimationHeader D_06008EEC;
+extern u64* D_06017410; // Title card
+extern AnimationHeader D_0601CAE0;
+extern AnimationHeader D_06008EEC;
+extern AnimationHeader D_060061D4;
+extern AnimationHeader D_0600DF38;
+extern AnimationHeader D_0600E848;
+extern AnimationHeader D_06004E0C;
+extern AnimationHeader D_060042A8;
+extern AnimationHeader D_06009D10;
+extern AnimationHeader D_0601D934;
+extern AnimationHeader D_06002D0C;
+extern AnimationHeader D_060042A8;
+extern AnimationHeader D_06001074;
+extern AnimationHeader D_06003CF8;
+extern s16 D_06015890[];
+extern s16 D_06017210[];
+extern s16 D_06015D90[];
+extern s16 D_06016390[];
+extern s16 D_06016590[];
+extern s16 D_06016790[];
+extern s16 D_06015990[];
+extern s16 D_06015F90[];
+extern s16 D_06016990[];
+extern s16 D_06016E10[];
+extern Gfx* D_06009D50[];
+extern Gfx* D_06009DD0[];
+
+void func_808C1190(s16* arg0, u8* arg1, s16 arg2) {
+    if (arg2[arg1] != 0) {
+        arg0[arg2 / 2] = 0;
+    }
+}
+
+void func_808C11D0(s16* arg0, u8* arg1, s16 arg2) {
+    if (arg1[arg2] != 0) {
+        arg0[arg2] = 0;
+    }
+}
+
+void func_808C1200(s16* arg0, u8* arg1, s16 arg2) {
+    if (arg1[arg2] != 0) {
+        arg0[arg2] = 0;
+    }
+}
+
+void func_808C1230(s16* arg0, u8* arg1, s16 arg2) {
+    s16 index;
+
+    if (arg1[arg2] != 0) {
+        index = ((arg2 & 0xF) + ((arg2 & 0xF0) * 2));
+        arg0[index + 16] = 0;
+        arg0[index] = 0;
+    }
+}
+
+void func_808C1278(s16* arg0, u8* arg1, s16 arg2) {
+    s16 index;
+
+    if (arg1[arg2] != 0) {
+        index = ((arg2 & 0xF) * 2) + ((arg2 & 0xF0) * 2);
+        arg0[index + 1] = 0;
+        arg0[index] = 0;
+    }
+}
+
+void func_808C12C4(void* arg1, s16 arg2) {
+    func_808C1190(SEGMENTED_TO_VIRTUAL(D_06015890), arg1, arg2);
+    func_808C1200(SEGMENTED_TO_VIRTUAL(D_06017210), arg1, arg2);
+    func_808C11D0(SEGMENTED_TO_VIRTUAL(D_06015D90), arg1, arg2);
+    func_808C11D0(SEGMENTED_TO_VIRTUAL(D_06016390), arg1, arg2);
+    func_808C11D0(SEGMENTED_TO_VIRTUAL(D_06016590), arg1, arg2);
+    func_808C11D0(SEGMENTED_TO_VIRTUAL(D_06016790), arg1, arg2);
+    func_808C1230(SEGMENTED_TO_VIRTUAL(D_06015990), arg1, arg2);
+    func_808C1230(SEGMENTED_TO_VIRTUAL(D_06015F90), arg1, arg2);
+    func_808C1278(SEGMENTED_TO_VIRTUAL(D_06016990), arg1, arg2);
+    func_808C1278(SEGMENTED_TO_VIRTUAL(D_06016E10), arg1, arg2);
+}
+
+void func_808C1554(u64* D_030021D8, u64* floorTex, s32 arg2, f32 arg3) {
+    u16* temp_s3 = SEGMENTED_TO_VIRTUAL(D_030021D8);
```

Comment:
I suggest using `void*` for these texture pointers, and the argument being named `D_030021D8` seems confusing so I think it should be named `arg0` if we don't know what to name it.
```suggestion
void func_808C1554(void* arg0, void* floorTex, s32 arg2, f32 arg3) {
    u16* temp_s3 = SEGMENTED_TO_VIRTUAL(arg0);
```